scripts/api_diff_report/api_info.py

```diff
@@ -0,0 +1,243 @@
+            }
+
+        module_api_container[api_type] = {
+            'api_type_link': api_type_link,
+            'apis': apis
+        }
+
+    parse_api(api_doc_path, module_api_container)
+
+    return module_api_container
+
+
+def parse_api(api_doc_path, module_api_container):
+    """Parse API html and extract necessary information.
+
+    e.g. ${module}/Classes.html
+    """
+    for api_type, api_type_abstract in module_api_container.items():
+        api_type_link = f'{api_doc_path}/{unquote(api_type_abstract["api_type_link"])}'
+        api_data_container = module_api_container[api_type]['apis']
+        with open(api_type_link, 'r') as file:
+            html_content = file.read()
+
+        # Parse the HTML content
+        soup = BeautifulSoup(html_content, 'html.parser')
+        for api in soup.find('div',
+                             class_='task-group').find_all('li',
+                                                           class_='item'):
+            api_name = api.find('a', class_='token').text
+            for api_declaration in api.find_all('div', class_='language'):
+                api_declaration_text = ' '.join(
+                    api_declaration.stripped_strings)
+                api_declaration_text = remove_commentary_lines(
+                    api_declaration_text)
+                api_data_container[api_name]['declaration'].append(
+                    api_declaration_text)
+
+        for api, api_abstruct in api_type_abstract['apis'].items():
+            if api_abstruct['api_link'].endswith('.html'):
+                parse_sub_api(
+                    f'{api_doc_path}/{unquote(api_abstruct["api_link"])}',
+                    api_data_container[api]['sub_apis'])
+
+
+def parse_sub_api(api_link, sub_api_data_container):
+    """Parse SUB_API html and extract necessary information.
+
+    e.g. ${module}/Classes/${class_name}.html
+    """
+    with open(api_link, 'r') as file:
+        html_content = file.read()
+
+    # Parse the HTML content
+    soup = BeautifulSoup(html_content, 'html.parser')
+    for s_api in soup.find('div', class_='task-group').find_all('li',
+                                                                class_='item'):
+        api_name = s_api.find('a', class_='token').text
+        sub_api_data_container[api_name] = {'declaration': []}
+        for api_declaration in s_api.find_all('div', class_='language'):
+            api_declaration_text = ' '.join(api_declaration.stripped_strings)
+            api_declaration_text = remove_commentary_lines(
+                api_declaration_text)
+            sub_api_data_container[api_name]['declaration'].append(
+                api_declaration_text)
+
+
+def remove_commentary_lines(declaration):
+    """This is a *PATCH*, that remove commentary lines from API info e.g.
+    Declaration of FIRAppCheck: https://firebase.google.com/docs/reference/ios/
+    firebaseappcheck/api/reference/Classes/FIRAppCheck."""
```

Comment:
@sunmou99, the AppCheck docs bug is fixed by #11224. Did you find any others? Thanks for surfacing this.

Reply:
This is the only place I find.

Reply:
You can remove this function as it is no longer needed.